pkg/webhook/resources/resourcequota/mutator.go

```diff
@@ -0,0 +1,92 @@
+package resourcequota
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ctlharvcorev1 "github.com/harvester/harvester/pkg/generated/controllers/core/v1"
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+const (
+	resourceQuotaHard         = "/spec/hard"
+	harvesterLoadBalancersKey = "count/loadbalancers.loadbalancer.harvesterhci.io"
+)
+
+func NewMutator(resourceQuotaCache ctlharvcorev1.ResourceQuotaCache) types.Mutator {
+	return &resourceQuotaMutator{
+		resourceQuotaCache: resourceQuotaCache,
+	}
+}
+
+// podMutator injects Harvester settings like http proxy envs and trusted CA certs to system pods that may access
+// external services. It includes harvester apiserver and longhorn backing-image-data-source pods.
```

Comment:
This comment seems irrelevant. We can update/remove it.

Reply:
Removed it. Thanks.